flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/filesystem/FileSystemTableSinkTest.java

```diff
@@ -0,0 +1,233 @@
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.filesystem;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.table.api.DataTypes;
+import org.apache.flink.table.api.EnvironmentSettings;
+import org.apache.flink.table.api.ExplainDetail;
+import org.apache.flink.table.api.TableConfig;
+import org.apache.flink.table.api.TableSchema;
+import org.apache.flink.table.api.config.ExecutionConfigOptions;
+import org.apache.flink.table.catalog.CatalogTableImpl;
+import org.apache.flink.table.catalog.ObjectIdentifier;
+import org.apache.flink.table.connector.ChangelogMode;
+import org.apache.flink.table.connector.sink.DataStreamSinkProvider;
+import org.apache.flink.table.connector.sink.DynamicTableSink;
+import org.apache.flink.table.descriptors.DescriptorProperties;
+import org.apache.flink.table.factories.FactoryUtil;
+import org.apache.flink.table.planner.utils.TableTestUtil;
+import org.apache.flink.table.planner.utils.TestingTableEnvironment;
+import org.apache.flink.table.runtime.connector.sink.SinkRuntimeProviderContext;
+import org.apache.flink.types.RowKind;
+import org.apache.flink.util.ExceptionUtils;
+
+import org.junit.Test;
+
+import scala.Option;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+/** Test for {@link FileSystemTableSink}. */
+public class FileSystemTableSinkTest {
+
+    private static final TableSchema TEST_SCHEMA =
+            TableSchema.builder()
+                    .field("f0", DataTypes.STRING())
+                    .field("f1", DataTypes.BIGINT())
+                    .field("f2", DataTypes.BIGINT())
+                    .build();
+
+    @Test
+    public void testFileSystemTableSinkWithParallelismInChangeLogMode() {
+        int parallelism = 2;
+        DescriptorProperties descriptor = new DescriptorProperties();
+        descriptor.putString(FactoryUtil.CONNECTOR.key(), "filesystem");
+        descriptor.putString("path", "/tmp");
+        descriptor.putString("format", "testcsv");
+        descriptor.putString(FactoryUtil.SINK_PARALLELISM.key(), String.valueOf(parallelism));
+
+        final DynamicTableSink tableSink =
+                FactoryUtil.createTableSink(
+                        null,
+                        ObjectIdentifier.of("mycatalog", "mydb", "mytable"),
+                        new CatalogTableImpl(TEST_SCHEMA, descriptor.asMap(), ""),
+                        new Configuration(),
+                        Thread.currentThread().getContextClassLoader(),
+                        false);
+        assertTrue(tableSink instanceof FileSystemTableSink);
+
+        final DynamicTableSink.SinkRuntimeProvider provider =
+                tableSink.getSinkRuntimeProvider(new SinkRuntimeProviderContext(false));
+        assertTrue(provider instanceof DataStreamSinkProvider);
+
+        try {
+            tableSink.getChangelogMode(
```

Comment:
I think it would be better to verify the exception using plain SQL instead of manually calling the special method.